nengo/utils/lock.py

```diff
@@ -0,0 +1,51 @@
+import errno
+import os
+import os.path
+import time
+
+
+class TimeoutError(Exception):
```

Comment:
Though, maybe it is useful to backends that need to acquire some hardware resource.